src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -18,6 +18,39 @@ public partial class MemoryMappedFile
             HandleInheritability inheritability, MemoryMappedFileAccess access,
             MemoryMappedFileOptions options, long capacity)
         {
+            bool isSpecial = false;
+
+            if (fileStream != null)
+            {
+                if (fileStream.Length == 0)
+                {
+                    // If the file does not have length, it may be special unix device.
+                    Interop.Sys.FileStatus status;
+                    if (Interop.Sys.FStat(fileStream.SafeFileHandle, out status) == 0)
+                    {
+                        if ((status.Mode & Interop.Sys.FileTypes.S_IFCHR) != 0)
```

Comment:
Totally optional - if you think it's just as easy to read, this `if` can be simplified:

```cs
isSpecial = (status.Mode & Interop.Sys.FileTypes.S_IFCHR) != 0;
```

Reply:
updated.